### PR TITLE
feat: adds support for ascii doc preview in readme

### LINKED ISSUE #464: Add support for README in AsciiDoc
We have all documentation, including project READMEs, written in AsciiDoc format. Verdaccio does not support AsciiDoc and tries to render package READMEs as Markdown, which results in wrongly formatted “package detail”.

Suggestion: Use [asciidoctor.js](https://github.com/asciidoctor/asciidoctor.js) to render files with file extension `.adoc` and `.asciidoc`.

### PR DESCRIPTION
**Pull request by ayusharma**
**Type:**

The following has been addressed in the PR:

*  There is a related issue #464 
*  Unit or Functional tests are included in the PR

**Description:**
- Integrates asciidoctor.js to add the asciidoc preview for readme.

### CODE REVIEW

File: src/lib/utils.js [hunk 1]
```diff
@@ -10,6 +10,9 @@ import _ from 'lodash';
 import createError from 'http-errors';
 import type {Package} from '@verdaccio/types';
 import type {$Request} from 'express';
+import marked from 'marked';
+// $FlowFixMe
+import asciidoctor from 'asciidoctor.js';
```
Comment (juanpicado) [inline src/lib/utils.js]: You can use here create-stub 
https://github.com/flowtype/flow-typed/wiki/CLI-Commands-and-Flags

Comment (juanpicado) [reply src/lib/utils.js]: Or try to find the lib whether exist

File: test/unit/utils.spec.js [hunk 2]
```diff
@@ -110,4 +110,14 @@ describe('Utilities', () => {
       expect(convertDist.versions['1.0.1'].dist.tarball).toEqual(buildURI(host, '1.0.1'));
     });
   });
+
+  describe('parseReadme', () => {
+    const markdown = '# markdown';
+    const ascii = "= AsciiDoc";
+
+    test('convert ascii/makrdown to html template', () => {
```
[the remark this answers is not in the document]
Comment (ayusharma) [reply test/unit/utils.spec.js]: Done 👍

File: src/lib/utils.js [hunk 3]
```diff
@@ -437,6 +440,22 @@ function addGravatarSupport(pkgInfo: any) {
...
+ * parse package readme - markdown/ascii
+ * @param {String} redme package readme
+ * @return {String} readme html template
+ */
+function parseReadme(readme: string): string {
+  const ascii = asciidoctor();
+  const docTypeIdentifier = new RegExp(/^=+ \w/, 'g');
+
+  if (docTypeIdentifier.test(readme)) {
+    return ascii.convert(readme, {safe: 'safe', attributes: {showtitle: true, icons: 'font'}});
```
[the remark this answers is not in the document]
Comment (ayusharma) [reply src/lib/utils.js]: we convert ascii first and if it is not ascii, we convert to markdown (html).

**Pull request merged**